src/Twig/Extensions.php

```diff
@@ -303,15 +303,17 @@ public function transformXml(string $xml, string $xsl)
      *
      * @return string
      */
-    public static function formatBytes($bytes, int $precision = 2) : string
+    public static function formatBytes($bytes, int $precision = 2, $unit = null) : string
     {
         if (empty($bytes)) {
             $bytes = 0;
         }
         $units = array('B','KB','MB','GB','TB');
         for ($e = (count($units) - 1); $e > 0; $e--) {
             $one = pow(1000, $e);
-            if ($bytes >= $one) {
+            if (!empty($unit) and $units[$e] == $unit) {
+                return round(($bytes / $one), $precision) . ' ' . $units[$e];
```

Comment:
I don't think it's returning the units label appended to the result, but I think returning the units with the number is a good idea anyway. Can this . ' ' . $units[$e] be removed altogether?

Reply:
That is what this function does, is to humanize to byte size, but here it's not really being used like that. Instead we could use divide the number of bytes by 10^12 and round it. You think that's better but this scenario?

Reply:
I think it would be more straightforward, but probably not worth changing at this point. It's just not that big a deal imho.